*** cdrecord/cdrecord.c.orig	Sun Mar  2 14:40:34 1997
--- cdrecord/cdrecord.c	Fri May 16 15:07:11 1997
***************
*** 248,253 ****
--- 248,254 ----
  	silent++;		/* Be quiet if this fails   */
  	if (!is_yamaha())	/* Yamaha cannot load disks */
  		scsi_load_unload(1);
+ 	scsi_start_stop_unit(1);
  	silent--;
  	silent++;
  	for (i=0; i < 60 && test_unit_ready() < 0; i++)
***************
*** 858,868 ****
--- 859,874 ----
  LOCAL void
  unload_media()
  {
+ #ifdef	NONONO
+ 	/*
+ 	 * Philips CDD-2600 don't like this.
+ 	 */
  	scsi_start_stop_unit(0);
  	silent++;
  	while (test_unit_ready() < 0)
  		sleep(1);
  	silent--;
+ #endif
  	scsi_prevent_removal(0);
  	scsi_load_unload(0);
  }
