*** defaults.h.orig	Tue Jun  2 04:40:37 1998
--- defaults.h	Sat Jun  6 17:38:21 1998
***************
*** 7,13 ****
  
  #define  PREPARER_DEFAULT 	NULL
  #define  PUBLISHER_DEFAULT	NULL
! #define  APPID_DEFAULT 		NULL
  #define  COPYRIGHT_DEFAULT 	NULL
  #define  BIBLIO_DEFAULT 	NULL
  #define  ABSTRACT_DEFAULT 	NULL
--- 7,15 ----
  
  #define  PREPARER_DEFAULT 	NULL
  #define  PUBLISHER_DEFAULT	NULL
! #ifndef	APPID_DEFAULT
! #define  APPID_DEFAULT 		"MKISOFS ISO 9660 FILESYSTEM BUILDER"
! #endif
  #define  COPYRIGHT_DEFAULT 	NULL
  #define  BIBLIO_DEFAULT 	NULL
  #define  ABSTRACT_DEFAULT 	NULL
***************
*** 24,30 ****
  #endif
  
  #ifdef __sun
! #ifdef __svr4__
  #define  SYSTEM_ID_DEFAULT    "Solaris"
  #else
  #define  SYSTEM_ID_DEFAULT    "SunOS"
--- 26,32 ----
  #endif
  
  #ifdef __sun
! #ifdef __SVR4
  #define  SYSTEM_ID_DEFAULT    "Solaris"
  #else
  #define  SYSTEM_ID_DEFAULT    "SunOS"
