*** write.c.last	Fri Jun  5 01:28:07 1998
--- write.c	Sat Jun  6 16:59:43 1998
***************
*** 130,140 ****
  	 * If the system cannot do this reliable, you need to use this
  	 * hacky option.
  	 */
! 	if (split_output != 0 && ftell(file) > (1024 * 1024 * 1024) ) {
! 		static int	idx = 0;
! 		char	nbuf[128];
  
! 		sprintf(nbuf, "part_%02d", idx++);
  		file = freopen(nbuf, "w", file);
  		if (file == NULL) {
  			fprintf(stderr, "Cannot open '%s'.\n", nbuf);
--- 130,144 ----
  	 * If the system cannot do this reliable, you need to use this
  	 * hacky option.
  	 */
! 	static	int	idx = 0;
! 	if (split_output != 0 &&
! 	    (idx == 0 || ftell(file) >= (1024 * 1024 * 1024) )) {
! 			char	nbuf[512];
! 		extern	char	*outfile;
  
! 		if (idx == 0)
! 			unlink(outfile);
! 		sprintf(nbuf, "%s_%02d", outfile, idx++);
  		file = freopen(nbuf, "w", file);
  		if (file == NULL) {
  			fprintf(stderr, "Cannot open '%s'.\n", nbuf);
