*** joliet.c.bak	Mon Feb 16 18:56:55 1998
--- joliet.c	Sat Mar 28 18:18:00 1998
***************
*** 707,713 ****
     * Do not split a directory entry across a sector boundary 
     */
    s_entry = this_dir->jcontents;
!   this_dir->ce_bytes = 0;
    for(s_entry = this_dir->jcontents; s_entry; s_entry = s_entry->jnext)
      {
        int jreclen;
--- 707,716 ----
     * Do not split a directory entry across a sector boundary 
     */
    s_entry = this_dir->jcontents;
! /*
!  * XXX Is it ok to comment this out?
!  */
! /*XXX JS  this_dir->ce_bytes = 0;*/
    for(s_entry = this_dir->jcontents; s_entry; s_entry = s_entry->jnext)
      {
        int jreclen;
