*** mkisofs.8.bak	Wed Feb 18 05:50:46 1998
--- mkisofs.8	Sat Mar 28 17:46:01 1998
***************
*** 54,59 ****
--- 54,62 ----
  .I preparer
  ]
  [
+ .B \-print-size
+ ]
+ [
  .B \-P
  .I publisher
  ]
***************
*** 333,338 ****
--- 336,350 ----
  .I \&.mkisofsrc
  with PREP=.
  If specified in both places, the command line version is used.
+ .TP
+ .I \-print-size
+ Print estimated filesystem size and exit. This option is needed for
+ Disk At Once mode and with some CD-R drives when piping directly into
+ .B cdrecord.
+ In this case it is needed to know the size of the filesustem before the
+ actual CD-creation is done.
+ The option \-print-size allows to get this size from a "dry-run" before
+ the CD is actually written.
  .TP
  .I \-R
  Generate SUSP and RR records using the Rock Ridge protocol to further describe
