*** multi.c.bak	Sat Dec  6 22:05:05 1997
--- multi.c	Sat Mar 28 18:08:57 1998
***************
*** 751,756 ****
--- 751,758 ----
    unsigned int ttbl_extent = 0;
    unsigned int ttbl_index  = 0;
  
+   char whole_path[1024];
+ 
    /*
     * Whatever is leftover in the list needs to get merged back
     * into the directory.
***************
*** 761,767 ****
  	{
  	  continue;
  	}
!       
        if( pnt[i]->name != NULL
  	  && strcmp(pnt[i]->name, "<translation table>") == 0 )
  	{
--- 763,781 ----
  	{
  	  continue;
  	}
! 
!       if( pnt[i]->name != NULL && pnt[i]->whole_name == NULL)
!        {
!          /*
!           * Set the name for this directory.
!           */
!          strcpy(whole_path, this_dir->de_name);
!          strcat(whole_path, SPATH_SEPARATOR);
!          strcat(whole_path, pnt[i]->name);
! 
!          pnt[i]->whole_name = strdup(whole_path);
!        }
! 
        if( pnt[i]->name != NULL
  	  && strcmp(pnt[i]->name, "<translation table>") == 0 )
  	{
***************
*** 909,914 ****
--- 923,931 ----
    strcat(whole_path, dpnt->name);
    this_dir->de_name = strdup(whole_path);
  
+   this_dir->whole_name = strdup(whole_path);
+   this_dir->dir_flags = DIR_HAS_DOT | DIR_HAS_DOTDOT;
+ 
    /*
     * Now fill this directory using information from the previous
     * session.
***************
*** 940,945 ****
--- 957,971 ----
  	{
  	  memset(contents[i]->isorec.extent, 0, 8);
  	}
+ 
+       /*
+        * Set the whole name for this file.
+        */
+       strcpy(whole_path, this_dir->whole_name);
+       strcat(whole_path, SPATH_SEPARATOR);
+       strcat(whole_path, contents[i]->name);
+ 
+       contents[i]->whole_name = strdup(whole_path);
  
        contents[i]->next = this_dir->contents;
        contents[i]->filedir = this_dir;
